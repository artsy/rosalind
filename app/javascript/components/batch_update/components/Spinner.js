import React from 'react'
import PropTypes from 'prop-types'
import styled, { keyframes } from 'styled-components'

const spinning = keyframes`
  from {
    transform: rotate(0deg)
  }
  to {
    transform: rotate(360deg)
  }
`

const Overlay = styled.div`
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
  height: calc(100vh - 50px);
  background: hsla(0, 0%, 100%, 0.90);
  display: flex;
  flex-direction: row;
  justify-content: center;
  z-index: 1
`

const AnimatedLine = styled.div`
  width: 12%;
  height: 1%;
  background: ${props => props.color};
  align-self: center;
  z-index: 2;
  animation: ${spinning} 2s linear infinite;
`

AnimatedLine.propTypes = {
  color: PropTypes.string
}

AnimatedLine.defaultProps = {
  color: 'black'
}

const Spinner = () => (
  <Overlay>
    <AnimatedLine />
  </Overlay>
)

export default Spinner
