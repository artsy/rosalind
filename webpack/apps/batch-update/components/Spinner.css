.Spinner__Overlay {
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
  height: calc(100vh - 50px);
  background: hsla(0, 0%, 100%, 0.90);
  display: flex;
  flex-direction: row;
  justify-content: center;
  z-index: 1
}

.Spinner__Line {
  width: 12%;
  height: 1%;
  background: black;
  align-self: center;
  animation: spin 2s linear infinite;
  z-index: 2;
}
