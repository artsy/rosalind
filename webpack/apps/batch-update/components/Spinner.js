import React from 'react'

export default function Spinner () {
  const overlayStyle = {
    position: 'absolute',
    left: 0,
    top: 0,
    right: 0,
    height: 'calc(100vh - 50px)',
    background: 'hsla(0, 0%, 100%, 0.90)',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    zIndex: 1
  }
  const barStyle = {
    width: '12%',
    height: '1%',
    background: 'black',
    alignSelf: 'center',
    animation: 'spin 2s linear infinite',
    zIndex: 2
  }
  return (
    <div style={overlayStyle}>
      <div style={barStyle} />
    </div>
  )
}
