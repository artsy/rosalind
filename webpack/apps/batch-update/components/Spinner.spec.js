import React from 'react'
import renderer from 'react-test-renderer'
import Spinner from './Spinner'

it('renders correctly', () => {
  const rendered = renderer.create(<Spinner />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})
