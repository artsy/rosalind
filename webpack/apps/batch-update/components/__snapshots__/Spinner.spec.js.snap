exports[`test renders correctly 1`] = `
<div
  className="lnzNIy">
  <div
    className="eWOvlO"
    color="black" />
</div>
`;
